packages/react-devtools-shared/src/utils.js

```diff
@@ -463,6 +464,10 @@ export function getDataType(data: Object): DataType {
     return 'undefined';
   }
 
+  if (data.self === data && data.parent && data.top) {
+    return 'window';
+  }
```

Comment:
This seems like an convoluted way to check for window objects. Why wouldn't something like `data instanceof Window` work?

Reply:
I'm also a little confused about why this fix (for window elements) is required to fix the linked bug about iframes. Wouldn't we need to be adding a check for `data instanceof HTMLIFrameElement` here instead of window?

Reply:
Hello @bvaughn 

> This seems like an convoluted way to check for window objects. Why wouldn't something like data instanceof Window work?

It doesn't always work for Iframe it considers that sometimes it is an Object. 
I've been looking at how the others are dealing with it:
JQuery: https://api.jquery.com/jquery.iswindow/
Anglular: https://github.com/angular/angular.js/blob/master/src/Angular.js#L677
I added some check to make it more robust but i flow the same behaviour.

> I'm also a little confused about why this fix (for window elements) is required to fix the linked bug about iframes. Wouldn't we need to be adding a check for data instanceof HTMLIFrameElement here instead of window?

I tested with the iframe element. It works

Reply:
I'm not sure I understand. This check doesn't really look like the two you linked to (JQuery and Angular). It also looks like it could hit a false positive, e.g.
```js
// This would be mistake as type "window"
const fake = {
  parent: 1,
  top: 1,
  // ...
};
fake.self = fake;
```
But it wouldn't if the check was:
```js
if (data instanceof Window || data instanceof HTMLIFrameElement) {
  return 'window';
}
```
So I'm curious why this doesn't work?

The JQuery function you linked to has been deprecated since v3.3 which is almost 3 years old, and the Angular check you linked to looks like it's in Angular v1 which is also deprecated.

Reply:
Okay I think I've come around to the necessity of this "looks like a window" check, although I think we need an inline comment here explaining the reason we check this way (aka the `iframe.contentWindow` case).

Maybe we could also handle the more common case explicitly? I dunno, maybe something like this?
```js
// We fallback to duck typing to handle an edge case:
// An HTMLIFrameElement's "contentWindow" is not an instance of Window,
// but rather an instance of the iframe's own Window (which we can't reliably access).
if (
  data instanceof Window ||
  data instanceof HTMLIFrameElement ||
  (data.self === data && data.parent != null && data.top != null)
) {
  return "window";
}
```

Also, maybe this whole conditional should be moved down into the `case 'object':` block.